### PR TITLE
Use java.lang.StringBuilder

### LINKED ISSUE #188: Use java.lang.StringBuilder
Scala's stringbuilder is a wrapper around java.lang.StringBuilder with largely the same methods but worse performance.

Performance focused code should use java.lang.StringBuilder

### PR DESCRIPTION
**Pull request by isaacl**
Prefer native StringBuilder over the scala wrapper.
This allows better hotspot optimization as well as
faster interpreter code.

Fixes #188.

### CODE REVIEW

File: scalatags/shared/src/main/scala/scalatags/Escaping.scala [hunk 1]
```diff
@@ -62,8 +64,7 @@ object Escaping {
         case '\n' => s.append('\n')
         case '\r' => s.append('\r')
         case '\t' => s.append('\t')
-        case c if c < ' ' =>
-        case c => s.append(c)
+        case c => if (c >= ' ') s.append(c)
```
Comment (marcenuc) [inline scalatags/shared/src/main/scala/scalatags/Escaping.scala]: Why not the following?

```suggestion
        case c if c >= ' ' => s.append(c)
```

Comment (isaacl) [reply scalatags/shared/src/main/scala/scalatags/Escaping.scala]: Well, technically your version is a non-exhaustive match. It seems scala doesn't warn once you have a guarded case like this, though.

Comment (marcenuc) [reply scalatags/shared/src/main/scala/scalatags/Escaping.scala]: Fair point about the non-exhaustive match. I did not think about that.

**Comment by isaacl** [comment]
@lihaoyi ping...

**Comment by lihaoyi** [comment]
I think this has been superseded by these methods now going to a generic `java.io.Writer`; you can put it into a `StringBuilder` or anywhere else you may like

**Pull request closed without merging**